libraries/abstractions/platform/freertos/iot_network_udp.c

```diff
@@ -0,0 +1,544 @@
+        }
+
+        /* Clear the connection information. */
+        if( ( pNetworkConnection != NULL ) && ( ( *pNetworkConnection ) != NULL ) )
+        {
+            vPortFree( *pNetworkConnection );
+        }
+    }
+
+    IOT_FUNCTION_CLEANUP_END();
+}
+/*-----------------------------------------------------------*/
+
+IotNetworkError_t IotNetworkUdp_SetReceiveCallback( void * pConnection,
+                                                    IotNetworkReceiveCallback_t receiveCallback,
+                                                    void * pContext )
+{
+    IOT_FUNCTION_ENTRY( IotNetworkError_t, IOT_NETWORK_SUCCESS );
+
+    /* Cast network connection to the correct type. */
+    _networkConnection_t * pNetworkConnection = ( _networkConnection_t * ) pConnection;
+
+    if( pConnection == NULL )
+    {
+        IotLogWarn( "Invalid connection parameter" );
+        IOT_SET_AND_GOTO_CLEANUP( IOT_NETWORK_BAD_PARAMETER );
+    }
+
+    /* Set the receive callback and context. */
+    pNetworkConnection->receiveCallback = receiveCallback;
+    pNetworkConnection->pReceiveContext = pContext;
+
+    /* No flags should be set. */
+    configASSERT( xEventGroupGetBits( ( EventGroupHandle_t ) &( pNetworkConnection->connectionFlags ) ) == 0 );
+
+    /* Create task that waits for incoming data. */
+    if( xTaskCreate( _networkReceiveTask,
+                     "NetUdpRecv",
+                     IOT_NETWORK_RECEIVE_TASK_STACK_SIZE,
+                     pNetworkConnection,
+                     IOT_NETWORK_RECEIVE_TASK_PRIORITY,
+                     &( pNetworkConnection->receiveTask ) ) != pdPASS )
+    {
+        configASSERT( pNetworkConnection->receiveTask );
+        IotLogError( "Failed to create network receive task." );
+
+        status = IOT_NETWORK_SYSTEM_ERROR;
+    }
+
+    IOT_FUNCTION_EXIT_NO_CLEANUP();
+}
+/*-----------------------------------------------------------*/
+
+size_t IotNetworkUdp_SendTo( void * pConnection,
+                           const uint8_t * pMessage,
+                           size_t messageLength )
+{
+    size_t bytesSent = 0;
+    _networkConnection_t * pNetworkConnection = ( _networkConnection_t * ) pConnection;
+
+    if( pConnection == NULL )
+    {
+        IotLogWarn( "Invalid connection parameter" );
+    }
+    else
+    {
+        bytesSent = SOCKETS_SendTo( pNetworkConnection->socket,
+                                    ( void * ) pMessage,
+                                    messageLength,
+                                    0,
+                                    &pNetworkConnection->serverAddr,
+                                    sizeof( SocketsSockaddr_t ) );
+    }
+
+    return bytesSent;
+}
+/*-----------------------------------------------------------*/
+
+size_t IotNetworkUdp_ReceiveFrom( void * pConnection,
+                              uint8_t * pBuffer,
+                              size_t bytesRequested )
+{
+    size_t bytesReceived = 0;
```

Comment:
This is not needed after the ```while``` loop is removed. I think ```socketStatus``` has all the info thats needed.